### PR TITLE
Use unions to represent Postgres enum types

### LINKED ISSUE #65: Union types for enums
Slight buyers remorse from me (shame I didn't think of this a bit earlier!): an issue with using Typescript enums for database enums is that the types we generate in different files are incompatible. That is, Typescript considers them to be different types when in fact they represent the same underlying type.

This ends up feeling quite clumsy if you are passing otherwise compatible objects around because you end up having to convert enum types; `payloadType as string as PayloadTypeFromQuery2`.

We could instead use unions, so:

```typescript
export const enum PayloadType {
  Message = 'Message',
  Dynamite = 'Dynamite',
}
```

becomes:

```typescript
export type PayloadType = 'Message' | 'Dynamite';
```

This means that we would get type-safety on enums and type arguments referencing the same database enum would be compatible.

We would loose the runtime abilities of a enum doing this though. Some thoughts on options:

1. Switch entirely to unions
2. Generate string enums for each type but make the parameters/returns be of union type
3. Attempt to 'globalise' enums across generation of different files sharing a enum so that the enum definition is hoisted to some location where it can be imported by all queries that use that enum

I am leaning towards 1 being a better default. We can also give an option to generate enums (but maybe we should favour simplicity and drop support for now). It maintains generated file locality and mostly 'just works'. If someone needs a runtime enum or prefers the enum syntax (which does feel clearer and more conventional) they can always create one.

2 is an attempt at the best of both worlds, but it feels a bit weird generating enums that we don't actually use directly, it won't be obvious that you can use them, and you'll loose editor/IDE support.

3 might be cleanest for a codebase but feels like scope creep and without knowing/assuming a lot more about a project structure would be hard to do.

### PR DESCRIPTION
**Pull request by silasdavis**
This provides assignability between enum-mapped types from different
query files where Typescript enums would be incompatible.

Also harness up pacakges/example as a form of integration/smoke test in
CI.

fixes #65

### CODE REVIEW

File: packages/example/scripts/wait-for-postgres-then [hunk 1]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+RETRIES=${RETRIES:-15}
+
+err_file=$(mktemp)
+
+echo "Checking postgres is up..."
+until psql "postgres" -c "select 1" > /dev/null 2> "$err_file" || [ "$RETRIES" -eq 0 ]; do
```
Comment (adelsz) [inline packages/example/scripts/wait-for-postgres-then:8]: Might be better to use [pg_isready](https://www.postgresql.org/docs/11/app-pg-isready.html), which can check for the DB to be created

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -3,13 +3,22 @@ name: CI
...
+
+  test_integration:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - run: npm ci
+      - name: Bootstrap packages
+        run: npx lerna bootstrap && npm run build
+      - name: Run integration tests
+        run: npm test:integration
```
Comment (adelsz) [inline .github/workflows/main.yml:24]:
```suggestion
        run: npm run test:integration
```
Looks like CI is failing because of this

Comment (silasdavis) [reply .github/workflows/main.yml]: yeah too used to yarn... incidentally I think we should switch :)

File: packages/example/scripts/wait-for-postgres-then (hunk 1 above)
Comment (silasdavis) [reply packages/example/scripts/wait-for-postgres-then:8]: oh nice I never knew about that

**Pull request merged**